doc/spec/manifest.md

```diff
@@ -0,0 +1,249 @@
+    
+        Abstract application data in any schema, as described by the
+        `mediaType` field.
+    
+    Application data may be *any* JSON object, allowing this manifest format,
+    along with the generic dependencies list, to support any type of 
+    application that can be represented as a JSON configuration and a
+    collection of content-addressable blobs of data.
+
+- **`metadata`** *object*
+
+    Metadata may be *any* JSON object, allowing content creators to annotate
+    their application manifest with any additional labels beyond those already
+    defined by this manifest format. 
+
+
+### Example Manifest
+
+```json
+{
+    "schemaVersion": 2,
+    "createdAt": "2015-02-11T15:07:17.639302443-08:00",
+    "maintainers": [
+        {
+            "name": "Phillip Edward Nuñez",
+            "email": "pnunez@example.com",
+            "url": "https://example.com/pnunez/"
+        }
+    ],
+    "description": "Application to Simulate Turbo Encabulators",
+    "keywords": [
+        "foo",
+        "bar",
+        "baz"
+    ],
+    "website": "https://foo.bar.baz.example.com",
+    "licenses": [
+        "Apache-2.0"
+    ],
+    "dependencies": [
+        {
+            "mediaType": "application/vnd.docker.distribution.content.manifest.v1+json",
+            "size": 1024,
+            "repository": "isv.example.com/base/system",
+            "digests": {
+                "sha256": "b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7"
+            }
+        },
+        {
+            "mediaType": "application/vnd.docker.container.image.layer+x-gtar",
+            "size": 32654,
+            "digests": {
+                "sha256": "e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7fc331f"
+            }
+        },
+        {
+            "mediaType": "application/vnd.docker.container.image.layer+x-gtar",
+            "size": 16724,
+            "digests": {
+                "sha256": "3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b"
+            }
+        },
+        {
+            "mediaType": "application/vnd.docker.container.image.layer+x-gtar",
+            "size": 73109,
+            "digests": {
+                "sha256": "ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736"
+            }
+        },
+        {
+            "mediaType": "application/vnd.docker.distribution.content.manifest.v1+json",
+            "size": 3072,
+            "repository": "example.com/foo/secrets",
+            "encryption": {
+                "algorithm": "AES-256-CTR",
+                "key": "ea7371db2eb566ba01d34669eca30eaaf4470ab58372790d61cce282d6e6b44f"
+            },
+            "digests": {
+                "sha256": "6e951eaa1c77c20611874f6a9d13d69bbcd15d93ceb21c1e22cc7ae65e2e36ea"
+            }
+        }
+    ],
+    "application": {
+        "mediaType": "application/vnd.docker.container.image.params.v1+json",
+        "data": {
+            "user": "alice",
```

Comment:
Yup--that idea looks really good to me as well.  I guess with the manifest being this flexible, the question is where do we articulate/define the scope of media types and get agreement from, say, docker client, on the algorithm or flow of lookup for various types?  However, I assume that is all reasonable and just a matter of discussion and definition.